src/main/java/com/vention/stock_market_share/service/TwelveDataService.java

```diff
@@ -0,0 +1,31 @@
+package com.vention.stock_market_share.service;
+
+import com.vention.stock_market_share.exception.TwelveDataApiException;
+import com.vention.stock_market_share.interfaces.TwelveApiClient;
+import com.vention.stock_market_share.response.TwelveDataApiResponse;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TwelveDataService {
+    private final TwelveApiClient twelveApiClient;
+
+    public TwelveDataService(TwelveApiClient twelveApiClient) {
+        this.twelveApiClient = twelveApiClient;
+    }
+
+    public TwelveDataApiResponse searchStockBySymbol(String symbol) {
+        try {
+            return twelveApiClient.searchStockBySymbol(symbol);
+        } catch (TwelveDataApiException e) {
+            throw new TwelveDataApiException("Error while interacting with AlphaVantage API", e);
```

Comment:
we should not throw errors in the catch block, only logging.

Reply:
logger.error() is used instead of throwing exception